File: ZI2 2021/lab7/src/Subject.php
<?php
namespace Wojtek\Lab7;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
/**
 * @ORM\Entity
 * @ORM\Table(name="subject")
 */
class Subject{
/** 
 * @ORM\Id
 * @ORM\Column(name="id", type="integer")
 * @ORM\GeneratedValue(strategy="IDENTITY")
*/
protected $id;
 /**
 * @ORM\Column(name="name",type="string")
 */
protected $name;
/**
* @ORM\ManyToOne(targetEntity="Teacher",inversedBy="subjects", cascade={"persist"})
*/
private $teachers;
/**
* @ORM\ManyToMany(targetEntity="Classs",inversedBy="subjects", cascade={"persist"})

*/
protected $classs;
public function __construct()
{
    $this->classs = new ArrayCollection();
}

//   public function getClasss(){
//       return $this->classs;
//  }
    /**
     * @return mixed
     */
public function setClasss($classs){
    $this->classs = $classs;
}
    /**
     * @return mixed
     */
public function getTeachers(){
    return $this->teachers;
}
    /**
     * @return mixed
     */
public function setTeachers($teachers){
    $this->teachers = $teachers;
}
    /**
     * @return mixed
     */
public function getId(){
    return $this->id;
}
    /**
     * @return mixed
     */
public function setId($id){
    $this->id = $id;
}
    /**
     * @return mixed
     */
public function getName(){
    return $this->name;
}
    /**
     * @return mixed
     */
public function setName($name){
    $this->name = $name;
}
}
?>

File: ZI2 2021/lab5/Subject.java
import javax.persistence.*;
import java.util.List;

@Entity
public class Subject {
    @Id
    @GeneratedValue
    private Long id;
    private String name;
    @ManyToOne(cascade = CascadeType.ALL)
    private Teacher teachers;

    public Teacher getTeachers() {
        return teachers;
    }

    public void setTeachers(Teacher teachers) {
        this.teachers = teachers;
    }

    public List<Class> getClasss() {
        return classs;
    }

    public void setClasss(List<Class> classs) {
        this.classs = classs;
    }

    @ManyToMany(cascade = CascadeType.ALL)
    private List<Class> classs;
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
